pacakges/server/src/course.rs

```diff
@@ -150,6 +151,15 @@ fn contains_course_number(str: &str) -> bool {
     false
 }
 
+pub fn vec_to_map(vec: Vec<Course>) -> HashMap<u32, Course> {
+    HashMap::from_iter(
+        vec.clone()
```

Comment:
Why do we need to clone vec?
Do you use it after this function? if not let's move it.

Reply:
because the vec is used twice in this context, once for the keys (only the numbers are used, but they have to be moved) and once for the values